### DISCUSSION TITLE
Rate limited today

### DISCUSSION THREAD

**Discussion by NavBlue1** [llamafilm/tesla-http-proxy-addon#91] [General]
Has anyone been rate limited recently?

I just got rate limited today for the first time, I’ve been running this add on since mid Feb and haven’t changed anything recently. This addon has been awesome. 

I get the Tesla custom integration to poll every 10 seconds at day time then change the polling interval to 11 minutes at night. 

What is everyone else doing with their polling intervals?

**Comment by sgriessini** [comment]
Getting the same here. I've turned on debugging, now I read:
```
2024-05-09T09:18:37+02:00 [debug] Sending request to https://fleet-api.prd.eu.vn.cloud.tesla.com/api/1/vehicles/LRWYGCEK2PC540837/signed_command: {"routable_message":"MgIIAjoSEhDX7uTw/39km2b0Pig8qBjemgMQVnp28twVRMqidRNRkdo0sHJDCkEE4RWn4eMVv73qPitXd4nDGqCnzsdnNwQN8MFOFHdMGJDuh21n1XLfP2rNeazvdS1+kx2Y74Q3anz1g+zEh1Pcbw=="}
2024-05-09T09:18:37+02:00 [debug] Server returned 429: Too Many Requests: Retry later
```

**Comment by iainbullock** [comment]
The Tesla documentation https://developer.tesla.com/docs/fleet-api recommends:

**Avoid frequently polling device data**
Fleet API allows developers to get data from devices via API, these are meant to be used infrequently. The majority of these APIs have stringent rate limits in place for protection. To get data at regular interval use [Fleet Telemetry](https://developer.tesla.com/docs/fleet-api#fleet-telemetry) which provides a direct and custom data stream in real time.

It also sets out the API limits:

Device data limits | 1 API request / car / 5 min And 1M API requests / day.
Commands limits | 50 API requests / car / day And 500k API requests / day.
Wake limits | 5 API requests / car / hour.
Energy Device data limits | 100k API requests / day.
Energy Settings limits | 5 requests / energy site / day.
Capabilities | API accessLogin with TeslaTesla for Business access Enterprise charging APIInvoicesLimited Fleet Telemetry streaming.
Price | $0/week.
Subscription duration | Weekly, auto-renewing until canceled.
Availability | This subscription plan is a temporary plan that will be replaced with different options in 2024.

I'm not sure what the difference between Device data limits and Commands limits is, but it seems that the limits per car per day is quite low

**Reply by sgriessini** [reply]
I'm using the commands to adjust the charging to my solar power production, sometimes a rate of 1 command per minute. Worked perfect until yesterday. Now the API seems to block after a few requests. With these limitations, my automation will not work anymore. Is this something recently activated?

**Reply by NavBlue1** [reply]
Sounds like rate limits are being enforced now.

I’m also using this for solar charging. Not sure if solar charging will be possible with these restrictions being enforced. 

Would love to hear from anyone still doing solar charging successfully and how their polling is setup.

**Comment by abreingan** [comment]
I also seem to be having this problem. 
I am using this for solar charging (potentially every 2 minutes) and it normally works well until sometime in the afternoon and then stops. 
In the Tesla Custom Integration there are also discussions on this including one by zlymeda who is using BLE via a modified proxy.
Is anyone else using this or have any pointers as to how?

**Reply by Briochorama** [reply]
Adding BLE capabilities to this module could indeed be a solution for us that charge our cars with excess solar power. @llamafilm, is there any way to do this ? As mentioned @zlymeda [seems to be using a modified version of the proxy](https://github.com/alandtse/tesla/issues/961#issuecomment-2114684006).

I don't want to create a mess in the repo. Is this something you'd wish to add to the proxy and, if so, should we open a new issue with the tag _enhancement_ ?

**Comment by bdeluca** [comment]
I also have the same solar requirement.

**Comment by apildk** [comment]
Finally got this working (using also for changing amps to adjust for solar every 30 sec) and then a short time later then i am getting this:
2024-08-10T11:25:18+02:00 [debug] Server returned 429: Too Many Requests: Retry in 52482 seconds

**Comment by mosshr** [comment]
> 2024-08-15: Self-service Fleet Telemetry
> 
> Fleet Telemetry onboarding no longer requires submitting a CSR to Tesla. Vehicles running firmware version 2024.26.4 or later can receive configurations through the vehicle-command HTTP proxy. Full setup details are available on the [Fleet Telemetry](https://developer.tesla.com/docs/fleet-api/fleet-telemetry) page.
> 
> Applications with CSRs submitted but not processed should migrate to using the new vehicle-command approach.
> 
> Applications configured using a CSR will continue to work, but are encouraged to migrate to the new process.

Ref: https://developer.tesla.com/docs/fleet-api/support/announcements

I found the announcement above which might resolve our issue (using our server to send commands directly to a Tesla), but I'm not knowledgeable enough. Does anyone have an idea if this will work?
